File: 2017/lua/day02.lua
local stuff = require("stuff")

local function findanddivide(t)
    for i = 1, #t - 1 do
        for j = i + 1, #t do
            if t[i] % t[j] == 0 then return t[i] / t[j] end
            if t[j] % t[i] == 0 then return t[j] / t[i] end
        end
    end
end

local checksum1, checksum2 = 0, 0
for line in io.lines("../__in/02.txt") do
    local nums = stuff.str2numtable(line)
    local limits = stuff.tableminmax(nums)
    checksum1 = checksum1 + limits[2] - limits[1]
    checksum2 = checksum2 + findanddivide(nums)
end

print(string.format("Part 1: %d", checksum1))
print(string.format("Part 2: %d", checksum2))

File: 2017/kotlin/day02.kt
import java.io.File

fun findPairAndDivide(list: List<Int>): Int {
    for (i in 0..list.lastIndex - 1) {
        for (j in (i + 1)..list.lastIndex) {
            if (list[j] % list[i] == 0) return list[j] / list[i]
        }
    }
    return 0
}

fun main() {
    val input = File("../__in/02.txt").readLines()
    
    var checksum1 = 0
    var checksum2 = 0
    
    for (line in input) {
        val list = line.split("""\s+""".toRegex()).map { it.toInt() }.sorted()
        checksum1 += list.last() - list.first()
        checksum2 += findPairAndDivide(list)
    }
    
    println("Part 1: $checksum1")
    println("Part 2: $checksum2")
}